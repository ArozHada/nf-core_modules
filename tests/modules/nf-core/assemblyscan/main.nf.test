nextflow_process {

    name "Test Process ASSEMBLYSCAN"
    script "modules/nf-core/assemblyscan/main.nf"
    process "ASSEMBLYSCAN"
    config "tests/modules/nf-core/assemblyscan/nextflow.config"
    tag "assemblyscan"

    test("Sarscov2 Genome Fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                             file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true) 
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
