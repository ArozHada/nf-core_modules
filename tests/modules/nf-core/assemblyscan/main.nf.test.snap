{
    "Sarscov2 Genome Fasta": {
        "content": [
            [
                "test.json:md5,9140e3d43f2d676f62e1325ace5dd8bd",
                "versions.yml:md5,406f16dbbbf8f98709caafe49ca10e21"
            ]
        ],
        "timestamp": "2023-04-07T20:51:11+0000"
    }
}